pandas/core/indexes/base.py

```diff
@@ -5194,7 +5194,7 @@ def _from_join_target(self, result: np.ndarray) -> ArrayLike:
         """
         if isinstance(self.values, BaseMaskedArray):
             return type(self.values)(result, np.zeros(result.shape, dtype=np.bool_))
-        elif isinstance(self.values, ArrowExtensionArray):
+        elif isinstance(self.values, ArrowExtensionArray | StringArray):
```

Comment:
```suggestion
        elif isinstance(self.values, (ArrowExtensionArray, StringArray)):
```

Reply:
Interesting, that was pre-commit…

Reply:
are you sure? it shouldn't rewrite this
```console
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ cat t.py
def foo():
    if isinstance(a, (str, int)):
        print(True)

(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ pyupgrade t.py --py311-plus
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ echo $?
0
```

Reply:
No I had something else initially that was rewritten into the | version instead of the version we have now

Reply:
do you happen to remember what? rewriting this outside of type annotations looks like a bug, I didn't think it was possible

no worries if not

Reply:
I think it was something like

```
if isinstance(a, int) or isinstance(a, str):
    ...
```